save-frontend/src/main/kotlin/com/saveourtool/save/frontend/utils/CustomHooks.kt

```diff
@@ -260,6 +257,29 @@ fun useEventStream(
             }
         }
 
+/**
+ * [useState] modification in order to support default state value from [Props].
+ * Usually data passed with [Props] is not up-to-date yet and expected to be updated (e.g. when the response is received)
+ * In this case, [useStateFromProps] should update the state once on first [valueFromProps] change.
+ *
+ * @param valueFromProps value that somehow depends on variable from [Props]
+ * @return [StateInstance] of [valueFromProps] changing its value on first [valueFromProps] change
+ * @see [useState]
+ */
+fun <T : Any> useStateFromProps(valueFromProps: T): StateInstance<T> {
+    val state = useState(valueFromProps)
+    val onceWrapper = useOnceAction()
+    val reference = useRef(valueFromProps)
+    useEffect(valueFromProps) {
+        if (reference.current != valueFromProps) {
```

Comment:
probably better to
```suggestion
        onceWrapper {
            if (reference.current != valueFromProps) {
```
then we will have equal count of states\effects